tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -11,130 +11,217 @@
+ * to negative values because of alignment on starting values.
+ * Also deletion  of objects created in scope of class lifetime may decrease
+ * values computed on previous measure.
+ */
+class MemCheckPipeline {
+private:
+    std::array<long, MeasureValueMax> measures;            // current measures
+    std::array<long, MeasureValueMax> start_measures;      // measures before run (will be used as baseline)
+public:
+    /**
+     * @brief Constructs MemCheckPipeline object and
+     *        measure values to use as baseline
+     */
+    MemCheckPipeline() {
+        start_measures[VMRSS] = (long) getVmRSSInKB();
+        start_measures[VMHWM] = start_measures[VMRSS];
+        start_measures[VMSIZE] = (long) getVmSizeInKB();
+        start_measures[VMPEAK] = start_measures[VMSIZE];
+        start_measures[THREADS] = (long) getThreadsNum();
+
+        std::copy(start_measures.begin(), start_measures.end(), measures.begin());
+    }
+
+    /**
+     * @brief Destructs MemCheckPipeline object and prints the latest measurements
+     */
+    ~MemCheckPipeline() {
+        // message required for DB data upload
+        log_info("Current values of virtual memory consumption:");
+        print_measures();
+    }
+
+    /**
+     * @brief Measures values at the current point of time
+     */
+    void do_measures() {
+        measures[VMRSS] = (long) getVmRSSInKB();
+        measures[VMHWM] = (long) getVmHWMInKB();
+        measures[VMSIZE] = (long) getVmSizeInKB();
+        measures[VMPEAK] = (long) getVmPeakInKB();
+        measures[THREADS] = (long) getThreadsNum();     // TODO: resolve *-32295
+    }
+
+    /**
+     * @brief Returns measurements aligned on a baseline
+     */
+    std::array<long, MeasureValueMax> get_measures() {
+        std::array<long, MeasureValueMax> temp;
+        std::transform(std::begin(measures), std::end(measures), std::begin(start_measures), std::begin(temp),
+                       [](long measure, long start_measure) -> long {
+                           return measure - start_measure;
+                       });
+        return temp;
+    }
+
+    /**
+     * @brief Returns measurements as string separated within hardcoded delimiter
+     */
+    std::string get_measures_as_str() {
+        std::array<long, MeasureValueMax> temp = get_measures();
+        std::string str = std::to_string(*temp.begin());
+        for (auto it = temp.begin() + 1; it != temp.end(); it++)
+            str += MEMCHECK_DELIMITER + std::to_string(*it);
+        return str;
+    }
+
+    /**
+     * @brief Prints headers and corresponding measurements using hardcoded delimiter
+     */
+    void print_measures() {
+        log_info(util::get_measure_values_headers(MEMCHECK_DELIMITER));
+        log_info(get_measures_as_str());
+    }
+
+    /**
+     * @brief Measures values at the current point of time and prints immediately
+     */
+    void print_actual_measures() {
```

Comment:
print "[MEASURE]" as special identificator for search and provide unique name through function API to use as DB step name

Reply:
Added `upload_..._measures` functions to be responsible for it